x-pack/legacy/plugins/alerting/server/plugin.ts

```diff
@@ -119,6 +121,7 @@ export class Plugin {
       taskManager: plugins.taskManager,
       securityPluginSetup: plugins.security,
       encryptedSavedObjectsPlugin: plugins.encryptedSavedObjects,
+      isESOUsingEphemeralEncryptionKey: this.isESOUsingEphemeralEncryptionKey!,
```

Comment:
Should we move the logic from the alerts client and move it into `getAlertsClientWithRequest` function below? 🤔

Reply:
Seems like the answer would be yes.  Unless the encryption key can be set in the config "live" via the NP plugin live config bits.  In that case, we'd need to rebuild the alerts client at the same time, at which point it would be happy with the key.

Reply:
seconded 👍 
That makes more sense to me too

Reply:
The one holdback I have is that we expose the AlertsClient class (https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/alerting/index.ts#L14) so others who use the class directly wouldn't be protected anymore. 🤔 We should probably just expose the interface instead of the class.

Reply:
++

Reply:
Ya, exposing an interface would be better, eg we could have a "disabled" version of the class which just threw errors for all the API calls, which we returned in this case, instead of the "real" one.

But it also makes me wonder if there are some APIs we should actually allow.  `find()` and `get()` seem ok to me, and I'm wondering if that would be useful to have available in this case.  Probably doesn't matter, since you can't really do anything else at this point anyway without fixing the encryption key config ...